tictactoe.py

```diff
@@ -1,26 +1,63 @@
+playerX_moves = []
+player0_moves = []
+winningcombo_1 = [1, 2, 3]
+winningcombo_2 = [4, 5, 6]
+winningcombo_3 = [7, 8, 9]
+winningcombo_4 = [1, 4, 7]
+winningcombo_5 = [2, 5, 8]
+winningcombo_6 = [3, 6, 9]
+winningcombo_7 = [1, 5, 9]
+winningcombo_8 = [3, 5, 7]
+
+print("""
+Let’s play tic-tac-toe! The developer is a newbie python coder 🐍
+so for now, plz only type integers from the empty list. 😜
+""")
+
 def game():
-    print("Lets play! Only integers between 1-9 plz:")
-
-    for playermove in range(0,9):
-        playermove = int(input(">>> "))
-
-        if playermove == 1:
-            print("Top left corner, all right. ↖️")
-        if playermove == 2:
-            print("Top center. ⬆️")
-        if playermove == 3:
-            print("Top right corner, nice. ↗️")
-        if playermove == 4:
-            print("Left center. ⬅️")
-        if playermove == 5:
-            print("Yeah that’s a good one. 🆒")
-        if playermove == 6:
-            print("Right center. ➡️")
-        if playermove == 7:
-            print("Bottom left corner, good job. ↙️")
-        if playermove == 8:
-            print("Bottom center. ⬇️")
-        if playermove == 9:
-            print("Bottom right corner, cool. ↘️")
+
+    if len(empty_squares) > 1:
+
+        next_turn_is_x = True
+
+        if next_turn_is_x == True:
+            playermove = int(input("\nPlayer X: "))
+            empty_squares.remove(playermove)
+            playerX_moves.append(playermove)
+            print(">>> POSITIONS by X: ", playerX_moves)
+            print(">>> POSITIONS by 0: ", player0_moves)
+            print(">>> Empty ", empty_squares)
+            next_turn_is_x = False
+
+        if set(winningcombo_1).issubset(playerX_moves) or set(winningcombo_2).issubset(playerX_moves) or set(winningcombo_3).issubset(playerX_moves) or set(winningcombo_4).issubset(playerX_moves) or set(winningcombo_5).issubset(playerX_moves) or set(winningcombo_6).issubset(playerX_moves) or set(winningcombo_7).issubset(playerX_moves) or set(winningcombo_8).issubset(playerX_moves):
+            print(">>> Tic-tac-toe and 🥇  for X")
+            return
+
+        if next_turn_is_x == False:
+            playermove = int(input("\nPlayer 0: "))
+            empty_squares.remove(playermove)
+            player0_moves.append(playermove)
+            print(">>> POSITIONS by X: ", playerX_moves)
+            print(">>> POSITIONS by 0: ", player0_moves)
+            print(">>> Empty ", empty_squares)
+            next_turn_is_x = True
+
+        if set(winningcombo_1).issubset(player0_moves) or set(winningcombo_2).issubset(player0_moves) or set(winningcombo_3).issubset(player0_moves) or set(winningcombo_4).issubset(player0_moves) or set(winningcombo_5).issubset(player0_moves) or set(winningcombo_6).issubset(player0_moves) or set(winningcombo_7).issubset(player0_moves) or set(winningcombo_8).issubset(player0_moves):
+            print(">>> Tic-tac-toe and 🥇  for 0")
+            return
+
+        game()
```

Comment:
Did you consider using a `while`-loop instead of recursion for this? In general, I think that's what most people would expect to see here (in truth I had to look for this line to understand how the game didn't terminate after only 1 move).

Reply:
In an earlier version I used `while len(empty_squares) > 1` to count down the moves. 

LPTHW has taught to use `while` loops sparingly, so I thought it’d be an improvement to get rid of it. But maybe this is the type of use case where it _does_ make sense. 🤔

Reply:
Most of the time, the `while`-loop isn't necessary. But I don't think it's dangerous to use it. Thinking "while not done: do these things" often helps me reason about code. But it's always possible to revisit things and turn it into a different solution later. In this case, we can actually consider using `for turn in range(9):` since we know that there will be at most 9 turns (but that wouldn't work if we were playing connect 4 instead).